sdk/easm/azure-analytics-defender-easm/src/samples/java/com/azure/analytics/defender/easm/DiscoveryRunsSample.java

```diff
@@ -0,0 +1,80 @@
+import com.azure.analytics.defender.easm.models.DiscoSource;
+import com.azure.analytics.defender.easm.models.DiscoSourceKind;
+import com.azure.analytics.defender.easm.models.DiscoGroupData;
+import com.azure.core.util.Configuration;
+import com.azure.identity.InteractiveBrowserCredentialBuilder;
+
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * This sample demonstrates how to create and manage discovery runs in a workspace
+ *
+ *  Set the following environment variables before running the sample:
+ *     1) SUBSCRIPTION_ID - the subscription id for your resource
+ *     2) WORKSPACE_NAME - the workspace name for your resource
+ *     3) RESOURCE_GROUP - the resource group for your resource
+ *     4) REGION - the azure region your resource is in
+ *     5) HOSTS - a comma separated list of hosts you would like to run discovery on
+ *     6) DOMAINS - a comma separated list of hosts you would like to run discovery on
+ */
+public class DiscoveryRunsSample {
+    public static void main(String[] args) {
+
+        // To create an easmClient, you need your subscription ID, endpoint, and some sort of credential.
+        String subscriptionId = Configuration.getGlobalConfiguration().get("SUBSCRIPTION_ID");
+        String workspaceName = Configuration.getGlobalConfiguration().get("WORKSPACENAME");
+        String resourceGroupName = Configuration.getGlobalConfiguration().get("RESOURCEGROUPNAME");
+        String region = Configuration.getGlobalConfiguration().get("REGION");
+        String endpoint = "https://" + region + ".easm.defender.microsoft.com";
+        String discoveryGroupName = "Sample Disco";
+        String discoveryGroupDescription = "This is a sample description for a discovery group";
+
+        EasmClient easmClient = new EasmClientBuilder()
+            .endpoint(endpoint)
+            .subscriptionId(subscriptionId)
+            .workspaceName(workspaceName)
+            .resourceGroupName(resourceGroupName)
+            // For the purposes of this demo, I've chosen the InteractiveBrowserCredential but any credential will work.
+            .credential(new InteractiveBrowserCredentialBuilder().build())
+            .buildClient();
+
+        // In order to start discovery runs, we must first create a discovery group, which is a collection of known assets that we can pivot off of.
+        // These are created using the discoveryGroupsPut method
+
+        List<DiscoSource> seeds = Stream.concat(
+            Arrays.stream(Configuration.getGlobalConfiguration().get("HOSTS").split(","))
+                .map((host) -> new DiscoSource().setName(host).setKind(DiscoSourceKind.HOST)),
+            Arrays.stream(Configuration.getGlobalConfiguration().get("DOMAINS").split(","))
+                .map((domain) -> new DiscoSource().setName(domain).setKind(DiscoSourceKind.DOMAIN))
+        ).collect(Collectors.toList());
```

Comment:
Can we provide a sample of the List in sample code?

I am not sure whether customer would know what HOSTS and DOMAINS should be (they likely won't have them in their env variables).